lib/smart_properties.rb

```diff
@@ -109,7 +109,7 @@ class << self
     #
     def included(base)
       base.extend(ClassMethods)
-      base.extend(ModuleMethods) if base.is_a?(Module)
+      base.extend(ModuleMethods) unless base.is_a?(Class)
```

Comment:
Can you make this `... if base.class == Module` instead?

Reply:
We discussed this offline and decided that `base.instance_of?(Module)` would be equivalent to `base.class == Module` and cleaner:

```ruby
$ pry
[1] pry(main)> class Foo; end
=> nil
[2] pry(main)> Foo.instance_of?(Module)
=> false
[3] pry(main)> Foo.instance_of?(Class)
=> true
[4] pry(main)> module Bar; end
=> nil
[5] pry(main)> Bar.instance_of?(Module)
=> true
[6] pry(main)> Bar.instance_of?(Class)
=> false
```

However, it looks like both of them don't support the module builder pattern:
```ruby
[7] pry(main)> class Baz < Module
[7] pry(main)* end
=> nil
[8] pry(main)> Baz.new.instance_of?(Module)
=> false
[9] pry(main)> Baz.new
=> #<Baz:0x00007f854609de40>
[10] pry(main)> Baz.new.class
=> Baz
[11] pry(main)> Module === Baz.new
=> true
[12] pry(main)> Class === Baz.new
=> false
```
which is a pattern that we both felt should be supported.

Finally, we agreed that since `base` is guaranteed to be a `Module` as passed to the `included` method, just checking that it is not a `Class` is enough to ensure that it is a plain module or a module builder instance.